wasm/instruction.go

```diff
@@ -0,0 +1,398 @@
+	OpcodeI64Mul:            "i64.mul",
+	OpcodeI64DivS:           "i64.div_s",
+	OpcodeI64DivU:           "i64.div_u",
+	OpcodeI64RemS:           "i64.rem_s",
+	OpcodeI64RemU:           "i64.rem_u",
+	OpcodeI64And:            "i64.and",
+	OpcodeI64Or:             "i64.or",
+	OpcodeI64Xor:            "i64.xor",
+	OpcodeI64Shl:            "i64.shl",
+	OpcodeI64ShrS:           "i64.shr_s",
+	OpcodeI64ShrU:           "i64.shr_u",
+	OpcodeI64Rotl:           "i64.rotl",
+	OpcodeI64Rotr:           "i64.rotr",
+	OpcodeF32Abs:            "f32.abs",
+	OpcodeF32Neg:            "f32.neg",
+	OpcodeF32Ceil:           "f32.ceil",
+	OpcodeF32Floor:          "f32.floor",
+	OpcodeF32Trunc:          "f32.trunc",
+	OpcodeF32Nearest:        "f32.nearest",
+	OpcodeF32Sqrt:           "f32.sqrt",
+	OpcodeF32Add:            "f32.add",
+	OpcodeF32Sub:            "f32.sub",
+	OpcodeF32Mul:            "f32.mul",
+	OpcodeF32Div:            "f32.div",
+	OpcodeF32Min:            "f32.min",
+	OpcodeF32Max:            "f32.max",
+	OpcodeF32Copysign:       "f32.copysign",
+	OpcodeF64Abs:            "f64.abs",
+	OpcodeF64Neg:            "f64.neg",
+	OpcodeF64Ceil:           "f64.ceil",
+	OpcodeF64Floor:          "f64.floor",
+	OpcodeF64Trunc:          "f64.trunc",
+	OpcodeF64Nearest:        "f64.nearest",
+	OpcodeF64Sqrt:           "f64.sqrt",
+	OpcodeF64Add:            "f64.add",
+	OpcodeF64Sub:            "f64.sub",
+	OpcodeF64Mul:            "f64.mul",
+	OpcodeF64Div:            "f64.div",
+	OpcodeF64Min:            "f64.min",
+	OpcodeF64Max:            "f64.max",
+	OpcodeF64Copysign:       "f64.copysign",
+	OpcodeI32WrapI64:        "i32.wrap_i64",
+	OpcodeI32TruncF32S:      "i32.trunc_f32_s",
+	OpcodeI32TruncF32U:      "i32.trunc_f32_u",
+	OpcodeI32TruncF64S:      "i32.trunc_f64_s",
+	OpcodeI32TruncF64U:      "i32.trunc_f64_u",
+	OpcodeI64ExtendI32S:     "i64.extend_i32_s",
+	OpcodeI64ExtendI32U:     "i64.extend_i32_u",
+	OpcodeI64TruncF32S:      "i64.trunc_f32_s",
+	OpcodeI64TruncF32U:      "i64.trunc_f32_u",
+	OpcodeI64TruncF64S:      "i64.trunc_f64_s",
+	OpcodeI64TruncF64U:      "i64.trunc_f64_u",
+	OpcodeF32ConvertI32s:    "f32.convert_i32_s",
+	OpcodeF32ConvertI32U:    "f32.convert_i32_u",
+	OpcodeF32ConvertI64S:    "f32.convert_i64_s",
+	OpcodeF32ConvertI64U:    "f32.convert_i64u",
+	OpcodeF32DemoteF64:      "f32.demote_f64",
+	OpcodeF64ConvertI32S:    "f64.convert_i32_s",
+	OpcodeF64ConvertI32U:    "f64.convert_i32_u",
+	OpcodeF64ConvertI64S:    "f64.convert_i64_s",
+	OpcodeF64ConvertI64U:    "f64.convert_i64_u",
+	OpcodeF64PromoteF32:     "f64.promote_f32",
+	OpcodeI32ReinterpretF32: "i32.reinterpret_f32",
+	OpcodeI64ReinterpretF64: "i64.reinterpret_f64",
+	OpcodeF32ReinterpretI32: "f32.reinterpret_i32",
+	OpcodeF64ReinterpretI64: "f64.reinterpret_i64",
+}
+
+// InstructionName returns the instruction corresponding to this binary Opcode.
+// See https://www.w3.org/TR/wasm-core-1/#a7-index-of-instructions
+func InstructionName(oc Opcode) string {
```

Comment:
fyi I renamed the file to instruction.go as the binary opcode is just one column describing an instruction.